### PR TITLE
languages: Add clangd_extensions, a plugin to handle clangd LSP extensions

### PR DESCRIPTION
**Pull request by traxys**

### CODE REVIEW

File: plugins/languages/clangd_extensions.nix [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+in
+  with lib; let
+    borderOpt = let
+      bordersTy =
+        types.enum ["double" "rounded" "single" "shadow" "none"];
+    in
+      helpers.defaultNullOpts.mkNullable (types.either bordersTy (types.listOf bordersTy))
+      ''"none"'' "";
+  in {
+    options.plugins.clangd_extensions = {
```
Comment (pta2002) [inline plugins/languages/clangd_extensions.nix:17]: Can you rename this to `clangd-extensions`? Keeps consistency with the rest

File: plugins/languages/clangd_extensions.nix [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      enable = mkEnableOption "clangd_extensions, plugin implementing clangd LSP extensions";
+
+      package = mkOption {
```
Comment (pta2002) [inline plugins/languages/clangd_extensions.nix:20]: There's also `mkPackageOption` in the helpers now

**Review by pta2002 — CHANGES_REQUESTED** [review]

File: plugins/languages/clangd_extensions.nix (hunk 1 above)
Comment (traxys) [reply plugins/languages/clangd_extensions.nix:17]: Yeah I was not really sure to respect the case of the plugin or the other plugins, will do!

File: plugins/languages/clangd_extensions.nix (hunk 2 above)
Comment (traxys) [reply plugins/languages/clangd_extensions.nix:20]: Ah right, will edit

**Comment by traxys** [comment]
Should be done, sorry for the force push needed it to rebase on master

**Review by pta2002 — APPROVED** [review]

**Pull request merged**